AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -2236,6 +2238,31 @@ - (void)setDisplaySuspended:(BOOL)flag
   }
 }
 
+static const char *ASDisplayNodeDrawingPriorityKey = "ASDrawingPriority";
+
+- (void)setDrawingPriority:(NSUInteger)drawingPriority
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  ASDN::MutexLocker l(_propertyLock);
+  if (drawingPriority == ASDefaultDrawingPriority) {
+    _flags.hasCustomDrawingPriority = NO;
+    objc_setAssociatedObject(self, ASDisplayNodeDrawingPriorityKey, nil, OBJC_ASSOCIATION_ASSIGN);
+  } else {
+    _flags.hasCustomDrawingPriority = YES;
```

Comment:
@appleguy, I did it under assumption that most nodes would have default priority, only few others (i.e. images) would have different values (i.e. instead of labels having increased priority I decreased priority for just the image). 

Of course if you start setting priorities for more nodes, at some point ivar will become better choice than associated object. I'm not sure what the constraints are, if it is considered cheap to add 4 or 8 bytes to every node then let's do it :)